lib/optimus_prime/modules/persistence/listener.rb

```diff
@@ -0,0 +1,62 @@
+module OptimusPrime
+  module Modules
+    module Persistence
+      class Listener
+        attr_reader :base
+        delegate :operation, :load_job, to: :base
+
+        def initialize(base)
+          @base = base
+          @pipeline_name = nil
+          @operation_id = nil
+          @jobs = {}
+        end
+
+        def pipeline_started(pipeline)
+          @pipeline_name = pipeline.name
```

Comment:
Why did you set @pipeline_name in #initialize and #pipeline_started ?  I think no one uses it.